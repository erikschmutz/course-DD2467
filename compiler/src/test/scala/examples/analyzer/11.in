let g a b = a + 1;;
let f a = g a;;
f 1 'b'

(*====================
val g : int=>'b=>int
val f : int=>'b=>int
- :int
====================*)