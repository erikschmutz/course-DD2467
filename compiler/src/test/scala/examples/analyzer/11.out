AnalyzerResult(
    Integer(),
    Enviroment(
        List(
            EnviromentEntry(
                Identifier(g),
                Function(
                    Integer(),
                    Function(
                        Unknown('b),
                        Integer()
                    )
                )
            ),
            EnviromentEntry(
                Identifier(f),
                Function(
                    Integer(),
                    Function(
                        Unknown('b),Integer()
                    )
                )
            )
        )
    )
)